modules/module-mysql/test/src/MysqlTypeMappings.test.ts

```diff
@@ -0,0 +1,302 @@
+import { SqliteRow } from '@powersync/service-sync-rules';
+import { afterAll, describe, expect, test } from 'vitest';
+import { clearTestDb, TEST_CONNECTION_OPTIONS } from './util.js';
+import { eventIsWriteMutation, eventIsXid } from '@module/replication/zongji/zongji-utils.js';
+import * as common from '@module/common/common-index.js';
+import ZongJi, { BinLogEvent } from '@powersync/mysql-zongji';
+import { MySQLConnectionManager } from '@module/replication/MySQLConnectionManager.js';
+import { toColumnDescriptors } from '@module/common/common-index.js';
+
+describe('MySQL Data Types', () => {
+  const connectionManager = new MySQLConnectionManager(TEST_CONNECTION_OPTIONS, {});
+
+  afterAll(async () => {
+    await connectionManager.end();
+  });
+
+  async function setupTable() {
+    const connection = await connectionManager.getConnection();
+    await clearTestDb(connection);
+    await connection.query(`CREATE TABLE test_data (
+    tinyint_col TINYINT,
+    smallint_col SMALLINT,
+    mediumint_col MEDIUMINT,
+    int_col INT,
+    integer_col INTEGER,
+    bigint_col BIGINT,
+    float_col FLOAT,
+    double_col DOUBLE,
+    decimal_col DECIMAL(10,2),
+    numeric_col NUMERIC(10,2),
+    bit_col BIT(8),
+    boolean_col BOOLEAN,
+    serial_col SERIAL,
+    
+    date_col DATE,
+    datetime_col DATETIME(3),
+    timestamp_col TIMESTAMP(3),
+    time_col TIME,
+    year_col YEAR,
+
+    char_col CHAR(10),
+    varchar_col VARCHAR(255),
+    binary_col BINARY(16),
+    varbinary_col VARBINARY(256),
+    tinyblob_col TINYBLOB,
+    blob_col BLOB,
+    mediumblob_col MEDIUMBLOB,
+    longblob_col LONGBLOB,
+    tinytext_col TINYTEXT,
+    text_col TEXT,
+    mediumtext_col MEDIUMTEXT,
+    longtext_col LONGTEXT,
+    enum_col ENUM('value1', 'value2', 'value3'),
+    set_col SET('value1', 'value2', 'value3'),
+
+    json_col JSON,
+
+    geometry_col GEOMETRY,
+    point_col POINT,
+    linestring_col LINESTRING,
+    polygon_col POLYGON,
+    multipoint_col MULTIPOINT,
+    multilinestring_col MULTILINESTRING,
+    multipolygon_col MULTIPOLYGON,
+    geometrycollection_col GEOMETRYCOLLECTION
+    )`);
+
+    connection.release();
+  }
+
+  test('Number types mappings', async () => {
+    await setupTable();
+    await connectionManager.query(`
+INSERT INTO test_data (
+    tinyint_col,
+    smallint_col,
+    mediumint_col,
+    int_col,
+    integer_col,
+    bigint_col,
+ -- float_col,
```

Comment:
Why is float excluded?

Reply:
With the rounding errors that creep in with the float mapping one cannot test for exact values.
I initially tested it by checking if the value was within a given precision range, but that didn't feel very useful so I ended up removing it.
Its due to these precision errors that MySQL recommends users should rather use the double type.